### PR TITLE
Write out CIFTI time series and correlations as TSV files

### LINKED ISSUE #555: Supplement parcellated time series and connectivity CIFTI files with TSVs
## Summary

The atlas time series and connectivity matrices are currently stored as CIFTI files. We should change them to TSV files.

## Additional details

This stems from @kahinimehta's comment in https://github.com/PennLINC/xcp_d/issues/517#issuecomment-1265808370.
I've split it off into a separate issue because this is a file format change, rather than just a naming change.

### PR DESCRIPTION
**Pull request by tsalo**
Closes #555.

To do:

- [x] Update the file patterns to support the new files.
- ~~[ ] Determine what, if any, metadata we want to store in sidecar files.~~ I'll probably deal with this in the provenance PR.
- ~~[ ] Remove header and index from conmat files, and move them to a `_nodelabels` file. Probably at the top level of the derivatives.~~ Everything's easier with a header and index, so I'll leave it for now.
- [x] Update the outputs documentation.

## Changes proposed in this pull request

- Create a new function, `xcp_d.utils.utils.extract_ptseries`, that uses nibabel and pandas to extract the parcellated time series.
- Create a new function, `xcp_d.utils.fcon.compute_functional_connectivity`, that uses pandas to perform the pair-wise Pearson correlations on the time series TSV file. This is used by both CIFTI and NIFTI workflows now.
- Remove the now-unused CiftiCorrelation interface.
- Change the time series and correlation outputs in the CIFTI workflow to be TSVs.
- Split up the steps in the NiftiConnect interface, with the masking/time series extraction done in one function Node and the correlation done in another. Remove the now unused NiftiConnect interface.
- Ensure that nifti time series and conmat files are structured the same way as cifti ones. I.e., currently time series have column headers and conmats have both column headers and index columns.

### CODE REVIEW

**Comment by tsalo** [comment]
The TSV files will need to be modified a bit to be BIDS compliant. Especially the conmat file.

**Comment by tsalo** [comment]
@kahinimehta WDYT about the conmat structure? Currently, I've got the index column of node names and the rest of the column names are node names as well. Based on BEP017, it seems like that's not preferred.

- Do you think I should make a `_nodelabels.tsv` file?
- If so, what metadata should go in the `_nodelabels.json` file?
- If not, should I use the `NodeLabels` metadata field in the `_conmat.json` file?

File: docs/output.rst [hunk 1]
```diff
@@ -7,93 +7,96 @@ Outputs of xcp_d
...
+               }
 
-   b. Functional timeseries and connectivity matrices::
+     #.   Functional timeseries and connectivity matrices::
 
-        #Nifti
-        xcp_d/sub-${sub-id}/ses-${ses-id}/func/sub-${sub-id}_ses-${ses-id}_task-${taskname}_run-${run}_space-${space}_atlas-${atlasname}_desc-timeseries_bold.tsv
-        xcp_d/sub-${sub-id}/ses-${ses-id}/func/sub-${sub-id}_ses-${ses-id}_task-${taskname}_run-${run}_space-${space}_atlas-${atlasname}_desc-connectivity_bold.tsv
+               # Nifti
+               xcp_d/sub-${sub-id}/ses-${ses-id}/func/sub-${sub-id}_ses-${ses-id}_task-${taskname}_run-${run}_space-${space}_atlas-${atlasname}_desc-timeseries_bold.tsv
```
Comment (kahinimehta) [inline docs/output.rst:56]: Should these be in the conmat format?

**Comment by kahinimehta** [comment]
Conmat structure looks good... I'd honestly be happy leaving the header and index in and having the atlas info kinda in a similar structure as it is now, but if we wanted to go by the BEPS, this is probably the BEP we are looking at https://docs.google.com/document/d/1RxW4cARr3-EiBEcXjLpSIVidvnUSHE7yJCUY91i5TfM/edit#heading=h.4k1noo90gelw

File: docs/output.rst (hunk 1 above)
Comment (tsalo) [reply docs/output.rst:56]: Good catch. I'll fix that.

Comment (tsalo) [reply docs/output.rst:56]: Oh actually I think I address that in #553.

**Comment by tsalo** [comment]
I like that idea. The atlas BEP seems to have more detailed, and useful, recommendations for atlases than I thought.

Unfortunately, I needed to get the NIFTI-based timeseries and conmats into the same structure to get everything working. NIFTIs, of course, don't have the associated node names within the files, so I need to refactor things a bit to carry along that info throughout the process, which really requires standardized atlases. Basically, finishing up this PR will need to wait until we have the atlases standardized.

**Comment by tsalo** [comment]
I think the core changes are solid. I'll wait until the atlases are added to the Docker image before marking this ready for review though, so that I'm not overhauling the package data multiple times in consecutive repos.

**Comment by tsalo** [comment]
I will write out both TSVs and ptseries/pconn files, which means I'll need to redo a lot of the code here.

**Comment by tsalo** [comment]
Closing in favor of #785.

**Pull request closed without merging**